index.js

```diff
@@ -55,8 +55,8 @@ function sanitize(input, replacement, invalids) {
     throw new Error("The replacement string can't be part of options.additionalInvalidStrings or contain substrings which are part of options.additionalInvalidStrings!");
   }
   var sanitized = input;
+  var counter = 0;
```

Comment:
Nice fix with moving the declaration outside of the `while` loop -- at first I thought that lgtm bot was crazy but I guess it was right.

Do you have a test that validates this working to prevent that bug from regressing in the future?

Reply:
I've downloaded eslint right after realizing what I've done lol. Embarrassing :D

I added the test `illegal replacement function`, currently in line 159 in `test.js`. This should check if the exception is thrown :)